### PR TITLE
[kueuectl] Add create resourceflavor command.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add `kueuectl create resourceflavor` command.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
CLI: Added create resourceflavor command.
```

### CODE REVIEW

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+	var tolerations []corev1.Toleration
+	uniqueTolerations := make(map[corev1.TaintEffect]sets.Set[string])
+
+	for _, tolerationSpec := range spec {
+		newTolerations, err := parseToleration(tolerationSpec)
+		if err != nil {
+			return nil, err
+		}
+		// validate if toleration is unique by <key, effect>
+		if len(uniqueTolerations[newTolerations.Effect]) > 0 && uniqueTolerations[newTolerations.Effect].Has(newTolerations.Key) {
```
Comment (trasc) [inline cmd/kueuectl/app/create/create_resourceflavor.go:287]:
```suggestion
		if uniqueTolerations[newTolerations.Effect].Has(newTolerations.Key) {
```

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+			return nil, fmt.Errorf("duplicated tolerations with the same key and effect: %v", newTolerations)
+		}
```
Comment (trasc) [inline cmd/kueuectl/app/create/create_resourceflavor.go:289]: should the value be relevant and operator?

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+		tolerations = append(tolerations, newTolerations)
+	}
+
+	return tolerations, nil
+}
+
+// parseToleration parses a toleration from a string, whose form must be either
+// 'KEY=VALUE:EFFECT', 'KEY:EFFECT' or 'KEY'
+func parseToleration(str string) (corev1.Toleration, error) {
```
Comment (trasc) [inline cmd/kueuectl/app/create/create_resourceflavor.go:304]: I think the split by "=" and ":" can be done in a common function.

```
func split(s string) (key, value, efect string, error) {....}
```

File: keps/2076-kueuectl/README.md [hunk 4]
```diff
@@ -153,6 +154,22 @@ kueuectl create lq|localqueue lqname
...
+### Create ResourceFlavor
+
+Creates a ResourceFlavor with the given name.
+
+Format:
+
+```
+kueuectl create rf|resourceflavor rfname
+--node-labels key1=value1,key2=value2
+--taints=key1 value:NoSchedule,key2:NoExecute
```
Comment (trasc) [inline keps/2076-kueuectl/README.md:166]:
```suggestion
--taints key1=value:NoSchedule,key2:NoExecute
```

File: keps/2076-kueuectl/README.md [hunk 5]
```diff
@@ -153,6 +154,22 @@ kueuectl create lq|localqueue lqname
...
+--tolerations key1=value:NoSchedule,key2:NoExecute
```
Comment (trasc) [inline keps/2076-kueuectl/README.md:167]:
```suggestion
--tolerations key1=value:NoSchedule,key2:NoExecute,key3
```

maybe it helps to describe how the parsing works

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+}
+
+func NewResourceFlavorCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewResourceFlavorOptions(streams)
+
+	cmd := &cobra.Command{
+		Use: "resourceflavor NAME " +
+			"[--node-labels KEY=VALUE] " +
+			"[--taints KEY=VALUE:EFFECT] " +
+			"[--tolerations KEY=VALUE:EFFECT] " +
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/kueuectl/app/create/create_resourceflavor.go]: Thanks

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+	var taints []corev1.Taint
+	uniqueTaints := make(map[corev1.TaintEffect]sets.Set[string])
+
+	for _, taintSpec := range spec {
+		newTaint, err := parseTaint(taintSpec)
+		if err != nil {
+			return nil, err
+		}
+		// validate if taint is unique by <key, effect>
+		if len(uniqueTaints[newTaint.Effect]) > 0 && uniqueTaints[newTaint.Effect].Has(newTaint.Key) {
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/kueuectl/app/create/create_resourceflavor.go]: Done

File: cmd/kueuectl/app/create/create_resourceflavor.go (hunk 1 above)
Comment (mbobrovskyi) [reply cmd/kueuectl/app/create/create_resourceflavor.go]: Done

File: cmd/kueuectl/app/create/create_resourceflavor.go (hunk 3 above)
Comment (mbobrovskyi) [reply cmd/kueuectl/app/create/create_resourceflavor.go:332]: Done

File: keps/2076-kueuectl/README.md (hunk 4 above)
Comment (mbobrovskyi) [reply keps/2076-kueuectl/README.md]: Fixed

File: keps/2076-kueuectl/README.md (hunk 5 above)
Comment (mbobrovskyi) [reply keps/2076-kueuectl/README.md]: Good point. Thanks.

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+			return nil, fmt.Errorf("duplicated taints with the same key and effect: %v", newTaint)
+		}
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/kueuectl/app/create/create_resourceflavor.go]: Looks like no https://github.com/kubernetes-sigs/kueue/blob/main/pkg/webhooks/resourceflavor_webhook.go#L118

File: cmd/kueuectl/app/create/create_resourceflavor.go (hunk 2 above)
Comment (mbobrovskyi) [reply cmd/kueuectl/app/create/create_resourceflavor.go]: For tolerations we don't have this validation. So removed from cmd.

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+}
+
+func NewResourceFlavorCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewResourceFlavorOptions(streams)
+
+	cmd := &cobra.Command{
+		Use: "resourceflavor NAME " +
+			"[--node-labels KEY=VALUE] " +
+			"[--taints KEY=VALUE:EFFECT] " +
+			"[--taints KEY[=VALUE]:EFFECT] " +
```
Comment (alculquicondor) [inline cmd/kueuectl/app/create/create_resourceflavor.go:94]:
```suggestion
			"[--node-taints KEY[=VALUE]:EFFECT] " +
```

File: cmd/kueuectl/app/create/create_resourceflavor.go [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+			}
+			return o.Run(cmd.Context())
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	cmd.Flags().StringToStringVar(&o.NodeLabels, nodeLabelsFlagName, nil,
+		"Labels that associate the ResourceFlavor with Nodes that have the same labels.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedTaints, taintsFlagName, nil,
```
Comment (alculquicondor) [inline cmd/kueuectl/app/create/create_resourceflavor.go:117]:
```suggestion
	cmd.Flags().StringSliceVar(&o.NodeTaints, taintsFlagName, nil,
```

**Review by alculquicondor — COMMENTED** [review]
/approve
/hold for minor comments

**Comment by alculquicondor** [comment]
/hold cancel
(I'll leave the LGTM to @trasc)

**Pull request merged**